src/ControllerDispatcher.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Zap\Routing;
+
+use Zap\Routing\Exceptions\InvalidRouteException;
+
+/**
+ * Class ControllerCaller
+ * @package Zap\Routing
+ * @author Gabor Zelei
+ */
+class ControllerDispatcher
+{
+    /**
+     * @var Route
+     */
+    private $route;
+
+    /**
+     * @var mixed       // @todo make sure this is actually Zap\Common\Interfaces\IController once IF is available
+     */
+    private $controller;
+
+    /**
+     * ControllerCaller constructor.
+     * @param Route $route
+     * @throws InvalidRouteException
+     */
+    public function __construct(Route $route)
+    {
+        $this->route = $route;
+
+        try {
+            $this->initController();
+        } catch (InvalidRouteException $e) {
+            throw $e;
+        }
+    }
+
+    /**
+     * Calls a controller based on current config
+     * @param array $params
+     * @return mixed
+     */
+    public function dispatchController(array $params = [])
```

Comment:
Do you think we could enforce to return `Response` class or `IResponse` or something similar?